integration_tests/src/main/python/array_test.py

```diff
@@ -109,3 +110,20 @@ def main_df(spark):
         chk_val = df.select(col('a')[0].alias('t')).filter(~isnan(col('t'))).collect()[0][0]
         return df.select(array_contains(col('a'), chk_val))
     assert_gpu_and_cpu_are_equal_collect(main_df)
+
+@pytest.mark.skipif(is_before_spark_311(), reason="This will throw exception only in Spark 3.1.1+")
+@pytest.mark.parametrize('data_gen', array_gens_sample, ids=idfn)
+def test_get_array_item_ansi_fail(data_gen):
+    assert_gpu_and_cpu_error(lambda spark: unary_op_df(
+        spark, data_gen).select(col('a')[100]).collect(),
+                               conf={'spark.sql.ansi.enabled':True,
+                                     'spark.sql.legacy.allowNegativeScaleOfDecimal': True},
+                               error_message='java.lang.ArrayIndexOutOfBoundsException')
+
+@pytest.mark.skipif(not is_before_spark_311(), reason="This will throw exception only in Spark 3.1.1+")
```

Comment:
The reason looks like a copy and paste. Under this test is in not clear what it means. It might be nice to update both to say something like `"In Spark 3.1.1+ ANSI mode array index throws on out of range indexes"`

Reply:
Sorry I missed that. Looks good then all I have is this nit.